### PR TITLE
Fix a bug with pluralization of third party resources

### PR DESCRIPTION
**Pull request by brendandburns**
Fixes https://github.com/kubernetes/kubernetes/issues/25129

@kubernetes/sig-api-machinery 

[Analytics]()

### CODE REVIEW

File: pkg/master/master.go [hunk 1]
```diff
@@ -670,19 +677,20 @@ func (m *Master) InstallThirdPartyResource(rsrc *extensions.ThirdPartyResource)
...
-func (m *Master) thirdpartyapi(group, kind, version string) *apiserver.APIGroupVersion {
+func (m *Master) thirdpartyapi(group, kind, version, pluralResource string) *apiserver.APIGroupVersion {
 	resourceStorage := thirdpartyresourcedataetcd.NewREST(
 		generic.RESTOptions{Storage: m.thirdPartyStorage, Decorator: generic.UndecoratedStorage, DeleteCollectionWorkers: m.deleteCollectionWorkers}, group, kind)
 
 	apiRoot := makeThirdPartyPath("")
 
 	storage := map[string]rest.Storage{
-		strings.ToLower(kind) + "s": resourceStorage,
+		pluralResource: resourceStorage,
```
Comment (lavalamp) [inline pkg/master/master.go]: So we're relying on meta.KindToResource's pluralization code working for arbitrary strings? I guess that's an improvement to just appending an 's'. Should we instead allow them to put a pluralized version in the specification? @smarterclayton here's another great reason not to use plurals in our paths...

File: pkg/master/master.go [hunk 2]
```diff
@@ -655,7 +656,13 @@ func (m *Master) InstallThirdPartyResource(rsrc *extensions.ThirdPartyResource)
 	if err != nil {
 		return err
 	}
-	thirdparty := m.thirdpartyapi(group, kind, rsrc.Versions[0].Name)
+	plural, _ := meta.KindToResource(unversioned.GroupVersionKind{
```
Comment (lavalamp) [inline pkg/master/master.go]: Just checking: that second parameter is not an error, correct?

Comment (brendandburns) [inline pkg/master/master.go]: Second parameter is the 'singular' version of the resource.

File: pkg/master/master.go (hunk 1 above)
Comment (brendandburns) [inline pkg/master/master.go]: KindToResource is what the kubectl client code uses afaik, so at least its consistent?

**Comment by brendandburns** [comment]
comments responded to.

**Comment by lavalamp** [comment]
LGTM

**Comment by brendandburns** [comment]
@k8s-bot node test this please flake: #25439

**Comment by brendandburns** [comment]
@k8s-bot node e2e test this please flake: #25439

**Comment by brendandburns** [comment]
@k8s-bot test this please issue: IGNORE

(forcing the retry without an issue because the logs for the failures above are missing, not sure why)

**Comment by brendandburns** [comment]
@k8s-bot test this please issue: 25699

**Comment by brendandburns** [comment]
@k8s-bot e2e test this please issue: #24794

**Comment by a-robinson** [comment]
Is this bug worth fixing for 1.3? If so, please add the 1.3 milestone.

**Comment by brendandburns** [comment]
@lavalamp for the 1.3 call.  I believe it's worth fixing.

**Comment by a-robinson** [comment]
@lavalamp might not see this until Thursday. In his stead, I'll say this should be fixed for 1.3, and we can sort it out later if he disagrees.

**Comment by wojtek-t** [comment]
I agree with @a-robinson

**Comment by lavalamp** [comment]
Yes, in 1.3 is good, thanks!

**Pull request merged**